docs/code-style.md

```diff
@@ -0,0 +1,8 @@
+# General code style guidelines and trends of the project.
+
+- Avoid augmenting `model` package. Instead, create a package structure according to the domain. See example [here](https://github.com/okteto/okteto/tree/master/pkg/externalresource).
+- [Mock filesystem calls](https://www.notion.so/okteto/Mock-filesystem-calls-53bf480a892849e7bbd81245457dba20)
```

Comment:
I don't think people have access to notion pages. Maybe we can create sections and explain each of the points so the people doesn't need access to notion